core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushPredicateIntoTableScan.java

```diff
@@ -287,6 +285,23 @@ private boolean arePlansSame(FilterNode filter, TableScanNode tableScan, PlanNod
         return Optional.of(tableScan);
     }
 
+    // PushProjectionIntoTableScan is executed after AddExchanges and UseConnectorTableScanPartitioning.
```

Comment:
`PushProjectionIntoTableScan is executed after AddExchanges`
Is it? It does not look so

Reply:
It happens as one of `io/trino/sql/planner/PlanOptimizers.java:391` optimizer execution (see line `io/trino/sql/planner/PlanOptimizers.java:798`)